include/mlir/Transforms/Utils.h

```diff
@@ -66,12 +66,20 @@ class OpBuilder;
 //  extra operands, note that 'indexRemap' would just be applied to existing
 //  indices (%i, %j).
 //  TODO(bondhugula): allow extraIndices to be added at any position.
-bool replaceAllMemRefUsesWith(Value *oldMemRef, Value *newMemRef,
-                              ArrayRef<Value *> extraIndices = {},
-                              AffineMap indexRemap = AffineMap(),
-                              ArrayRef<Value *> extraOperands = {},
-                              Operation *domInstFilter = nullptr,
-                              Operation *postDomInstFilter = nullptr);
+LogicalResult replaceAllMemRefUsesWith(Value *oldMemRef, Value *newMemRef,
+                                       ArrayRef<Value *> extraIndices = {},
+                                       AffineMap indexRemap = AffineMap(),
+                                       ArrayRef<Value *> extraOperands = {},
+                                       Operation *domInstFilter = nullptr,
+                                       Operation *postDomInstFilter = nullptr);
+
+/// Performs the same replacement as the other version above but only for the
+/// deferencing uses of `oldMemRef` in `op`.
```

Comment:
I am not sure what  "deferencing uses of `oldMemRef` in `op`." means?

Reply:
A dereferencing use is one where the memref is dereferenced, i.e., an individual element of it is accessed. For eg:

%v =  load %A[%i, %j] : memref<100 x 100 x f32> 

is a deferencing use of %A. affine.load, affine.store, load, store, dma_start, dma_finish are examples of ops that have deferencing uses of the memref. OTOH, uses in ops like:

%v = call "foo" (%A) : memref<100 x 100 x f32>
dealloc %A : memref<100 x 100 x f32>

are all non-deferencing uses of the memref. The doc comment is merely meant as a indirect reference to the version of the method above, which has all the info.

Reply:
Sorry, just realized there was a typo! Thanks.

Reply:
Fixed.